code/player/Player.cs

```diff
@@ -58,7 +60,10 @@ public override void Spawn()
         Tags.Add("player");
     }
 
-    public override void ClientSpawn() { }
+    public override void ClientSpawn() 
+    {
+        HeadTag = new UI.NameTag(this);
```

Comment:
Instead of initializing `NameTag` here, you could make a method in Player.UI.cs called something like `InitializeUI()`, initialize `NameTag` inside of that method, and call that method from here. This way, we would pretty much completely avoid doing UI stuff in the main file of the Player class.